EventGenerator/src/RMCGenerator_tool.cc

```diff
@@ -45,14 +45,16 @@ namespace mu2e {
       _czmin(conf().spectrum.get<fhicl::ParameterSet>().get<double>("czmin", -1.)),
       _czmax(conf().spectrum.get<fhicl::ParameterSet>().get<double>("czmax",  1.)),
       _spectrum(BinnedSpectrum(conf().spectrum.get<fhicl::ParameterSet>())),
+      _useRate(conf().spectrum.get<fhicl::ParameterSet>().get<bool>("useRate",  false)),
+      _internalRate((_external) ? 0. : 1.),
```

Comment:
I’m struggling to follow the settings needed to get the following configurations
- external (photons) only
- internal conversions only
- physical admixture of internal and external
Some combination of the ‘external’ and ‘userate’ bools seems to provide these, but I can’t quite follow.
It would be clearer if the valid configuration options were explicitly enumerated (enum) and set in the config.

I also find the term ‘external’ for producing RMC photons unintuitive, as those photons might be detected directly in the calorimeter, in which case there’s nothing ‘external’ about them.

Reply:
Only one of the `external` or `useRate` flags are necessary. I updated the configuration to use validated fhicl and in the update I set the `external` flag to only be required if `useRate` is false. I could update this to use strings or enums to check for something like "natural", "external", or "internal" if this is a preferred logic. I also don't prefer `external` as I term (I typically write real photons or virtual photons/internal conversions in talks), but it better matches the process IDs used and alternate generators.

Reply:
I prefer the conceptually more clear config to choose between the 3 sensible options. The names you give above are clear. If the process names are unclear we can change them, but lets do that in a separate PR.